### PR TITLE
correct inheritance of platforms by rules from groups

### LINKED ISSUE #9482: CPE seems to be broken on group/rule level
#### Description of problem:
```
:: [ 10:24:59 ] :: [   PASS   ] :: Uninstall postfix package (Expected 0, got 0)
:: [ 10:24:59 ] :: [  BEGIN   ] :: Scanning postfix rules without installed postfix :: actually running 'oscap xccdf eval --progress /tmp/tmp.5BxKguE6rF/ssg-rhel8-xccdf.xml'
WARNING: Datastream component 'scap_org.open-scap_cref_security-data-oval-com.redhat.rhsa-RHEL8.xml.bz2' points out to the remote 'https://access.redhat.com/security/data/oval/com.redhat.rhsa-RHEL8.xml.bz2'. Use '--fetch-remote-resources' option to download it.
WARNING: Skipping 'https://access.redhat.com/security/data/oval/com.redhat.rhsa-RHEL8.xml.bz2' file which is referenced from datastream
WARNING: Skipping ./security-data-oval-com.redhat.rhsa-RHEL8.xml.bz2 file which is referenced from XCCDF content
xccdf_org.ssgproject.content_rule_postfix_client_configure_relayhost:notchecked
xccdf_org.ssgproject.content_rule_postfix_network_listening_disabled:pass
xccdf_org.ssgproject.content_rule_postfix_server_banner:fail
xccdf_org.ssgproject.content_rule_postfix_prevent_unrestricted_relay:fail
:: [ 10:25:04 ] :: [   PASS   ] :: Scanning postfix rules without installed postfix (Expected 0,2, got 2)
:: [ 10:25:04 ] :: [   FAIL   ] :: File '/var/tmp/rlRun_LOG.zPkDSqS2' should contain ':notapplicable' 
xccdf_org.ssgproject.content_rule_postfix_network_listening_disabled:pass
xccdf_org.ssgproject.content_rule_postfix_server_banner:fail
xccdf_org.ssgproject.content_rule_postfix_prevent_unrestricted_relay:fail
:: [ 10:25:04 ] :: [   FAIL   ] :: File '/var/tmp/rlRun_LOG.zPkDSqS2' should not contain '(:pass|:fail|:error)' 
```

Rule that should be evaluated as notapplicable are returning fail/pass.

Potentially related to https://github.com/ComplianceAsCode/content/pull/9465

### PR DESCRIPTION
**Pull request by vojtapolasek**
#### Description:

- correct the process of passing on platforms so that they do not appear in multiple places in the datastream
- update the tests_machine_only_rules tests which ensures that machine-only applicability propagates correctly from groups to rules. This has to be changed because now when checking, we rely on XCCDF processing to apply platforms correctly, while when remediating we rely on generated remediations containing correct conditionals.

#### Rationale:

Fixes #9482

### CODE REVIEW

File: tests/test_machine_only_rules.py [hunk 1]
```diff
@@ -47,31 +49,41 @@ def check_ds(ds_path, what, input_elems):
...
     root = tree.getroot()
-    if what == "Group":
-        replacement = "xccdf_org.ssgproject.content_group_"
-        xpath_query = ".//{%s}Group" % ssg.constants.XCCDF12_NS
-    if what == "Rule":
-        replacement = "xccdf_org.ssgproject.content_rule_"
-        xpath_query = ".//{%s}Rule" % ssg.constants.XCCDF12_NS
+    # if what == "Group":
+        # replacement = "xccdf_org.ssgproject.content_group_"
+        # xpath_query = ".//{%s}Group" % ssg.constants.XCCDF12_NS
```
Comment (matejak) [inline tests/test_machine_only_rules.py]: Please remove, commented-out code is evil.

File: tests/test_machine_only_rules.py [hunk 2]
```diff
@@ -21,58 +31,94 @@ def main():
...
         sys.exit(1)
 
+
 def check_product(build_dir, product, rules_dirs):
     input_groups, input_rules = scan_rules_groups(rules_dirs, False)
     ds_path = os.path.join(build_dir, "ssg-" + product + "-ds.xml")
-    if not check_ds(ds_path, "Group", input_groups):
-        return False
-    if not check_ds(ds_path, "Rule", input_rules):
+    if machine_platform_missing_in_rules(ds_path, input_rules):
```
Comment (jan-cerny) [inline tests/test_machine_only_rules.py:45]: The input_groups is unused now unused. Is that intentional?

File: tests/test_machine_only_rules.py [hunk 3]
```diff
@@ -21,58 +31,94 @@ def main():
...
         return False
     return True
```
Comment (jan-cerny) [inline tests/test_machine_only_rules.py:47]: could be simply `return not machine_platform)missing_in_rules(...)`

File: tests/test_machine_only_rules.py [hunk 4]
```diff
@@ -21,58 +31,94 @@ def main():
...
+    return fixes_by_system
+
+
+def machine_platform_missing_in_rules(ds_path, short_ids_to_check):
+    machine_platform_missing = False
+
     try:
         tree = ET.parse(ds_path)
-    except IOError as e:
+    except IOError:
```
Comment (jan-cerny) [inline tests/test_machine_only_rules.py:78]: If the data stream doesn't exist, it doesn't make sense to scan and load compiled rules for that product.

File: tests/test_machine_only_rules.py [hunk 5]
```diff
@@ -21,58 +31,94 @@ def main():
...
-        return False
-    if not check_ds(ds_path, "Rule", input_rules):
+    if machine_platform_missing_in_rules(ds_path, input_rules):
         return False
     return True
 
 
-def check_ds(ds_path, what, input_elems):
+def shorten_id(full_id):
+    id_prefix = "xccdf_org.ssgproject.content_rule_"
```
Comment (jan-cerny) [inline tests/test_machine_only_rules.py:51]: you can use OSCAP_RULE constant from ssg/constants.py

File: tests/test_machine_only_rules.py (hunk 2 above)
Comment (vojtapolasek) [reply tests/test_machine_only_rules.py:45]: Yes, this is intentional. We currently use groups only when reading the directory structure (rule.yamls, group.yamls) and determining if the rule is inheriting the "machine" platform. But we are not checking their platform attribute in XCCDF anymore.

Comment (jan-cerny) [reply tests/test_machine_only_rules.py]: so you can just replace `input_groups` by `_`

**Comment by vojtapolasek** [comment]
@jan-cerny  thanks for feedback, I addressed it.

File: tests/test_machine_only_rules.py (hunk 2 above)
Comment (vojtapolasek) [reply tests/test_machine_only_rules.py]: Correct, fixed. Thanks.

**Review by jan-cerny — APPROVED** [review]
I have checked that the rule postfix_network_listening_disabled becomes notapplicable again when postfix isn't installed.

**Pull request merged**